clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -252,6 +252,33 @@ public synchronized void update(Cluster newCluster, Set<String> unavailableTopic
         if (isClosed())
             throw new IllegalStateException("Update requested after metadata close");
 
+        String previousClusterId = cluster.clusterResource().clusterId();
```

Comment:
is it possible for cluster to be null? on 1st connection?

Reply:
No, the clusterId will be null on 1st connection, but the cluster object itself will always be non-null.